src/chainrules.jl

```diff
@@ -2,6 +2,48 @@ ChainRulesCore.@scalar_rule(xlogx(x::Real), (1 + log(x),))
 ChainRulesCore.@scalar_rule(xlogy(x::Real, y::Real), (log(y), x / y,))
 ChainRulesCore.@scalar_rule(xlog1py(x::Real, y::Real), (log1p(y), x / (1 + y),))
 
+function ChainRulesCore.frule((_, Δx), ::typeof(xexpx), x::Real)
+    e = exp(x)
+    if iszero(e)
+        Ω = e
+        ΔΩ = e * Δx
+    else
+        Ω = x * e
+        ΔΩ = (1 + x) * e * Δx
+    end
+    return Ω, ΔΩ
+end
+
+function ChainRulesCore.rrule(::typeof(xexpx), x::Real)
+    e = exp(x)
+    Ω = iszero(e) ? e : x * e
+    xexpx_pullback(Δ) = (ChainRulesCore.NoTangent(), iszero(e) ? e * Δ : (1 + x) * e * Δ)
```

Comment:
I'd suggest breaking up the pullback to increase readability and to rename `e`. E.g.,
```suggestion
    expx = exp(x)
    Ω = iszero(expx) ? expx : x * expx
    function xexpx_pullback(ΔΩ)
        Δx = iszero(expx) ? expx * ΔΩ : (1 + x) * expx * ΔΩ
        return ChainRulesCore.NoTangent(), Δx
    end
```